main.tf

```
@@ -153,65 +63,37 @@ module "log_storage" {
   context = module.this.context
 }
 
-resource "aws_s3_bucket" "default" {
-  count = local.bucket_enabled ? 1 : 0
-
-  #bridgecrew:skip=BC_AWS_S3_13:Skipping `Enable S3 Bucket Logging` check until Bridgecrew will support dynamic blocks (https://github.com/bridgecrewio/checkov/issues/776).
-  #bridgecrew:skip=CKV_AWS_52:Skipping `Ensure S3 bucket has MFA delete enabled` check due to issues operating with `mfa_delete` in terraform
-  bucket        = substr(local.bucket_name, 0, 63)
-  acl           = var.acl
-  force_destroy = var.force_destroy
-  policy        = local.policy
-
-  versioning {
-    enabled    = true
-    mfa_delete = var.mfa_delete
-  }
-
-  server_side_encryption_configuration {
-    rule {
-      apply_server_side_encryption_by_default {
-        sse_algorithm = "AES256"
-      }
-    }
-  }
+module "tfstate_s3_bucket" {
+  source  = "cloudposse/s3-bucket/aws"
+  version = "0.49.0"
 
-  dynamic "replication_configuration" {
-    for_each = var.s3_replication_enabled ? toset([var.s3_replica_bucket_arn]) : []
-    content {
-      role = aws_iam_role.replication[0].arn
-
-      rules {
-        id     = module.this.id
-        prefix = ""
-        status = "Enabled"
-
-        destination {
-          bucket        = var.s3_replica_bucket_arn
-          storage_class = "STANDARD"
-        }
-      }
-    }
-  }
+  enabled            = local.bucket_enabled
+  bucket_name        = local.bucket_name
+  acl                = var.acl
+  force_destroy      = var.force_destroy
+  versioning_enabled = true
+  sse_algorithm      = "AES256"
 
-  dynamic "logging" {
-    for_each = var.logging == null ? [] : [1]
-    content {
-      target_bucket = local.logging_bucket_name
-      target_prefix = local.logging_prefix
-    }
-  }
+  #bridgecrew:skip=CKV_AWS_52:Skipping `Ensure S3 bucket has MFA delete enabled` due to issue in terraform (https://github.com/hashicorp/terraform-provider-aws/issues/629).
+  #mfa_delete = var.mfa_delete
 
-  tags = module.this.tags
-}
+  allow_encrypted_uploads_only = local.prevent_unencrypted_uploads
+  allow_ssl_requests_only      = local.prevent_unencrypted_uploads
 
-resource "aws_s3_bucket_public_access_block" "default" {
-  count                   = local.bucket_enabled && var.enable_public_access_block ? 1 : 0
-  bucket                  = join("", aws_s3_bucket.default.*.id)
   block_public_acls       = var.block_public_acls
-  ignore_public_acls      = var.ignore_public_acls
   block_public_policy     = var.block_public_policy
+  ignore_public_acls      = var.ignore_public_acls
   restrict_public_buckets = var.restrict_public_buckets
+
+  s3_replication_enabled = var.s3_replication_enabled
+  s3_replica_bucket_arn  = var.s3_replica_bucket_arn
+
+  logging = var.logging == null ? null : {
+    bucket_name = local.logging_bucket_name
+    prefix      = local.logging_prefix
+  }
```

Can we frame this in the positive ?

```suggestion
  logging = var.logging =! null ? {
    bucket_name = local.logging_bucket_name
    prefix      = local.logging_prefix
  } : null
```